arelle/examples/plugin/validate/ABC/__init__.py

```diff
@@ -0,0 +1,150 @@
+        *args: Any,
+        **kwargs: Any,
+    ) -> tuple[tuple[str, str]]:
+        return ((DISCLOSURE_SYSTEM_VALIDATION_TYPE, DISCLOSURE_SYSTEM_TEST_PROPERTY),)
+
+    @staticmethod
+    def modelDocumentPullLoader(
+        modelXbrl: ModelXbrl,
+        normalizedUri: str,
+        filepath: str,
+        isEntry: bool,
+        namespace: str | None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> ModelDocument | LoadingException | None:
+        disclosureSystem = modelXbrl.modelManager.disclosureSystem
+        if not disclosureSystemSelected(disclosureSystem):
+            return None
+        if isEntry and "abc" in filepath:
+            return LoadingException(_("Entrypoint file must not contain 'abc' in file name."))
+        return None
+
+    @staticmethod
+    def modelXbrlLoadComplete(
+        modelXbrl: ModelXbrl,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        disclosureSystem = modelXbrl.modelManager.disclosureSystem
+        if not disclosureSystemSelected(disclosureSystem):
+            return None
+        if modelXbrl.modelDocument is None:
+            modelXbrl.error(
+                codes="ABC.01.01",
+                msg=_("An XBRL Report Package is required but could not be loaded"),
+                modelObject=modelXbrl,
+            )
+        return None
+
+    @staticmethod
+    def validateXbrlFinally(
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        disclosureSystem = val.disclosureSystem
+        if not disclosureSystemSelected(disclosureSystem):
+            return None
+        for rule in validationRules.getValidations(
+            ValidationHook.XBRL_FINALLY, disclosureSystem
+        ):
+            rule(val, *args, **kwargs)
+        return None
+
+    @staticmethod
+    def validateXbrlDtsDocument(
+        val: ValidateXbrl,
+        modelDocument: ModelDocument,
+        isFilingDocument: bool,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        disclosureSystem = val.disclosureSystem
+        if not disclosureSystemSelected(disclosureSystem):
+            return None
+        for rule in validationRules.getValidations(
+            ValidationHook.XBRL_DTS_DOCUMENT, disclosureSystem
+        ):
+            rule(val, modelDocument, isFilingDocument, *args, **kwargs)
+        return None
+
+    @staticmethod
+    def validateFinally(
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        disclosureSystem = val.disclosureSystem
+        if not disclosureSystemSelected(disclosureSystem):
+            return None
+        for rule in validationRules.getValidations(
+            ValidationHook.FINALLY, disclosureSystem
+        ):
+            rule(val, *args, **kwargs)
+        return None
+
+
+__pluginInfo__ = {
+    "name": "Validate ABC",
+    "version": "0.0.1",
+    "description": "Example validation plugin for the fictitious ABC taxonomy.",
```

Comment:
I would probably error on calling this validation plugin example or something just to be extra clear